net: dhcpv4: net_dhcpv4_stop() removing address during renew

vlotech opened the pull request:
Should net_dhcpv4_stop be called when the interface is
in NET_DHCPV4_RENEWING state, than then address is not removed.
By also deleting the address in the renewing state means the status is 
always equal to the moment before _start is called. 

Signed-off-by: Vincent van der Locht <vincent@vlotech.nl>

tbursztyka reviewed (changes requested):
Do no touch indentation of comments, or else. First, it has _nothing_ to do with you actual patch, and second: actual indentation was just fine (we enforce 80chars limite, a comment made in a case should stay there etc etc...)

vlotech committed:
net: dhcpv4: net_dhcpv4_stop() removing address during renew

In case "net_dhcpv4_stop(..) is called when the interface is
in NET_DHCPV4_RENEWING state, the address is not removed.
When deleting the address in the NET_DHCPV4_RENEWING state
means the status is always equal to the moment before _start
is called.

Signed-off-by: Vincent van der Locht <vincent@vlotech.nl>

vlotech commented:
reverted the changes made by the uncrustify scripts (executed as mentioned in the contributors guide)

jukkar reviewed (changes requested):
Could you rephrase the commit message a bit, I have hard time understanding what it says as it is not proper English?

vlotech commented:
Commit message is updated. Hopefully it's clear now. 
The proposed changes makes sure that the network interface returns to the same state, after doing a start followed by stop, the network interface will return in the original state. The "RENEW" state was not clearing specific settings. 
This patch does fix it.

jukkar reviewed (approved):
LGTM

vlotech commented:
@tbursztyka: Could you please update the review comments, as the incorrect changes have been reversed in the patch. My apologies for these unneeded changes.
